client/order/cancel-authorization-confirm-modal/index.tsx

```diff
@@ -0,0 +1,134 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import interpolateComponents from '@automattic/interpolate-components';
+import { useState } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import ConfirmationModal from 'wcpay/components/confirmation-modal';
+
+interface CancelAuthorizationConfirmationModalProps {
+	originalOrderStatus: string;
+}
+
+const CancelAuthorizationConfirmationModal: React.FunctionComponent< CancelAuthorizationConfirmationModalProps > = ( {
+	originalOrderStatus,
+} ) => {
+	const [
+		isCancelAuthorizationConfirmationModalOpen,
+		setIsCancelAuthorizationConfirmationModalOpen,
+	] = useState( true );
+
+	const closeModal = (): void => {
+		setIsCancelAuthorizationConfirmationModalOpen( false );
+	};
+
+	const handleCancelOrder = (): void => {
+		const orderEditForm: HTMLFormElement | null = document.querySelector(
+			'#post'
+		);
+		if ( null !== orderEditForm ) {
+			orderEditForm.submit();
+		}
+	};
+
+	const doNotCancel = (): void => {
+		const orderStatusElement: HTMLInputElement | null = document.querySelector(
+			'#order_status'
+		);
+		if ( null !== orderStatusElement ) {
+			orderStatusElement.value = originalOrderStatus;
+			orderStatusElement.dispatchEvent( new Event( 'change' ) );
+		}
+		closeModal();
+	};
+
+	const cancelOrder = (): void => {
+		handleCancelOrder();
+		closeModal();
+	};
+
+	const buttonContent = (
+		<>
+			<Button isSecondary onClick={ doNotCancel }>
+				{ __( 'Do Nothing', 'woocommerce-payments' ) }
+			</Button>
+			<Button isPrimary onClick={ cancelOrder }>
```

Comment:
An alternative approach could be to craft an href to cancel the order, in a similar way how the order actions in the preview list work. To be on the safe side, we could build the url in PHP side and expose to the JS through the usual global. Here we'd have the following code:

```suggestion
			<Button isPrimary href={ wcpay_order_config.cancelOrderUrl }>
```
Note that we would get rid of the `cancelOrder` function.

Reply:
It's a bit complicated, I think the current approach is better. I'm checking how to make it more reliable, will post a new comment soon.